### PR TITLE
fix(Component): simplify List header

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
In the previous version of List header, we has some accessibility issues. The `<th>`contains
* a label that is readable
* a label or a button in a div that was `aria-hidden`

The problem was that the button was in the `aria-hidden`div. So blind people could not clic on the button to sort the list.

**What is the new behavior?**
The `<th>` content is now 
* a label that is hidden and aria-hidden, but present for the column sizing
* a div that is displayed with the button

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

File: output/components.eslint.txt [hunk 1]
```diff
@@ -26,7 +26,10 @@ Scoping to packages that match 'react-talend-components'
   2:1  error  '@kadira/react-storybook-addon-info' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   7:2  error  Unexpected require()                                                                                      global-require
 
-✖ 11 problems (11 errors, 0 warnings)
+/home/travis/build/Talend/ui/packages/components/src/utils/getPropsFrom.js
+  27:1  error  Trailing spaces not allowed  no-trailing-spaces
```
Comment (frassinier) [inline output/components.eslint.txt]: Did you miss it?

File: packages/components/src/List/DisplayTable/DisplayTable.component.js [hunk 2]
```diff
@@ -117,33 +118,43 @@ function getNextDirection(isCurrentSortField, currentSort) {
...
+	if (sort) {
+		const isCurrentSortField = sort.field === column.key;
+		const onChange = event => sort.onChange(
+			event,
+			{
+				field: column.key,
+				isDescending: getNextDirection(isCurrentSortField, sort.isDescending),
+			},
+		);
+		const actionProps = {
```
Comment (frassinier) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: Where can I find a story with this option?

File: output/components.eslint.txt (hunk 1 above)
Comment (jsomsanith) [inline output/components.eslint.txt]: Oh, thank you, yes missed it :(

File: packages/components/src/List/DisplayTable/DisplayTable.component.js (hunk 2 above)
Comment (jsomsanith) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: Here : http://118.talend.surge.sh/components/?selectedKind=List&selectedStory=Table%20with%20sort%20header%20click&full=0&down=1&left=1&panelRight=0&downPanel=kadirahq%2Fstorybook-addon-actions%2Factions-panel

Comment (jsomsanith) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: BTW are you proud of me with this `aria-label` accessibility master ?

File: output/components.eslint.txt (hunk 1 above)
Comment (jsomsanith) [inline output/components.eslint.txt]: Fixed

File: packages/components/src/List/DisplayTable/DisplayTable.component.js [hunk 3]
```diff
@@ -117,33 +118,43 @@ function getNextDirection(isCurrentSortField, currentSort) {
...
+			iconTransform: getIconTransform(sort.isDescending),
+			label: column.label,
+			link: true,
+			onClick: onChange,
+		};
+
+		header = <Action {...actionProps} />;
+	} else {
+		header = <span className={theme.header}>{column.label}</span>;
+	}
```
Comment (talend-glorieux) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: Don't you think the header would be better of as its own functions.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

File: packages/components/src/List/DisplayTable/DisplayTable.component.js (hunk 3 above)
Comment (jsomsanith) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: Done

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

File: packages/components/src/List/DisplayTable/DisplayTable.component.js (hunk 2 above)
Comment (frassinier) [inline packages/components/src/List/DisplayTable/DisplayTable.component.js]: :tada: :tada: tada: tada: :tada:

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://118.talend.surge.sh)

**Pull request merged**